--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -19,8 +19,8 @@ find_package(LZ4 REQUIRED)
 find_package(WebP REQUIRED) # v1.3.2, spews an error on Linux because no pkg-config.
 find_package(SDL3 3.2.6 REQUIRED)
 find_package(Freetype 2.11.1 REQUIRED)
-find_package(plutovg REQUIRED) # v0.0.13 is needed for building plutosvg, but we can support v1.0.0
-find_package(plutosvg 0.0.6 REQUIRED)
+find_package(plutovg 1.1.0...<2.0.0 REQUIRED)
+find_package(plutosvg 0.0.7...<1.0.0 REQUIRED)
 
 if(USE_VULKAN)
 	find_package(Shaderc REQUIRED)
